l1-contracts/contracts/bridge/interfaces/IL1SharedBridge.sol

```diff
@@ -0,0 +1,136 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.20;
+
+import {L2TransactionRequestTwoBridgesInner} from "../../bridgehub/IBridgehub.sol";
+import {IBridgehub} from "../../bridgehub/IBridgehub.sol";
+import {IL1ERC20Bridge} from "./IL1ERC20Bridge.sol";
+
+/// @title L1 Bridge contract interface
+/// @author Matter Labs
+/// @custom:security-contact security@matterlabs.dev
+interface IL1SharedBridge {
+    event DepositInitiatedSharedBridge(
+        uint256 indexed chainId,
+        bytes32 indexed l2DepositTxHash,
+        address indexed from,
+        address to,
+        address l1Token,
+        uint256 amount
+    );
+
+    event BridgehubDepositInitiatedSharedBridge(
+        uint256 indexed chainId,
+        bytes32 indexed txDataHash,
+        address indexed from,
+        address to,
+        address l1Token,
+        uint256 amount
+    );
+
+    event BridgehubDepositFinalized(
+        uint256 indexed chainId,
+        bytes32 indexed txDataHash,
+        bytes32 indexed l2DepositTxHash
+    );
+
+    event WithdrawalFinalizedSharedBridge(
+        uint256 indexed chainId,
+        address indexed to,
+        address indexed l1Token,
+        uint256 amount
+    );
+
+    event ClaimedFailedDepositSharedBridge(
+        uint256 indexed chainId,
+        address indexed to,
+        address indexed l1Token,
+        uint256 amount
+    );
+
+    function isWithdrawalFinalized(
+        uint256 _chainId,
+        uint256 _l2BatchNumber,
+        uint256 _l2MessageIndex
+    ) external view returns (bool);
+
+    function depositLegacyErc20Bridge(
+        address _msgSender,
+        address _l2Receiver,
+        address _l1Token,
+        uint256 _amount,
+        uint256 _l2TxGasLimit,
+        uint256 _l2TxGasPerPubdataByte,
+        address _refundRecipient
+    ) external payable returns (bytes32 txHash);
+
+    function claimFailedDepositLegacyErc20Bridge(
```

Comment:
The `IL1SharedBridge` interface is used by the bridge hub during deposits. 

I think we should keep this interface as clean as possible and avoid introducing unnecessary methods (like legacy methods which will be specific only for era)

The context is I am looking to implement a custom L1 bridge that can be plugged to the bridgehub and I find myself having to create unnecessary methods in my contract just to comply to this interface

Reply:
We want to have an interface for the current L1Shared Bridge. If you are already implementing a custom bridge, then copying the interface and removing the legacy functions should not be a problem.